### PR TITLE
V0.0.2

### PR DESCRIPTION
**Pull request by hansvancalster**

### CODE REVIEW

**Comment by florisvdh** [comment]
The workflow also takes quite long because packages need compiling. You could take advantage of the Posit Package Manager to get the binary (precompiled) packages. To achieve that on my machine, I have 

```r
> getOption("repos")
                                                         RSPM 
"https://packagemanager.posit.co/cran/__linux__/jammy/latest" 
                                                         CRAN 
                                "https://cloud.r-project.org" 
                                                         INLA 
                  "https://inla.r-inla-download.org/R/stable" 
                                                         inbo 
                                "https://inbo.r-universe.dev"
```

(Even that is a fallback; I use the fabulous [r2u](https://eddelbuettel.github.io/r2u/) repo to get R-packages as system packages prepared for Ubuntu instead.)

Also, [`r-lib/actions/setup-r`](https://github.com/r-lib/actions/tree/v2-branch/setup-r) has a switch `use-public-rspm` to set PPM/RSPM in GHA.

**Comment by hansvancalster** [comment]
I do not immediately see how I should implement that in the GHA. The GHA check_on_branch / check_on_main calls a docker container (https://github.com/inbo/actions/blob/main/check_pkg/action.yml).
So probably something inside the docker container should change in order to make it use RSPM / PPM? @ThierryO ?

**Comment by florisvdh** [comment]
Beside the **checklist** notes/errors, also see the CRAN incoming note in `R CMD check`:

```
* checking CRAN incoming feasibility ... [6s/16s] NOTE
Maintainer: ‘Hans Van Calster <hans.vancalster@inbo.be>’
New submission
Package CITATION file contains call(s) to old-style citEntry().  Please
use bibentry() instead.
Found the following (possibly) invalid URLs:
  URL: https://www.ogc.org/standards/filter (moved to https://www.ogc.org/standard/filter/)
    From: man/get_feature_wfs.Rd
    Status: 200
    Message: OK
  URL: https://www.ogc.org/standards/wfs (moved to https://www.ogc.org/standard/wfs/)
    From: man/get_feature_wfs.Rd
    Status: 200
    Message: OK
```

The CITATION file and URL validity issues are best handled IMO.

File: DESCRIPTION
Comment (ThierryO) [inline DESCRIPTION:21]: Why not use the base pipe `|>` instead of the magrittr pipe `%>%`? Development on the latter is stopped.

File: R/crs.R
Comment (ThierryO) [inline R/crs.R:39]: Maybe write this as below. Some people might (incorrectly) think you do `par_deg / (180 * pi)`.
```suggestion
  par <- par_deg * pi / 180
```

File: R/crs.R
Comment (ThierryO) [inline R/crs.R:35]: Note that you can add information below to the `authors` section of `_pkgdown.yml`

```
  Flors Vanderhaeghe:
    href: "https://github.com/florisvdh"
```

File: R/get_coverage_wcs.R
Comment (ThierryO) [inline R/get_coverage_wcs.R:23]: You can specify the linter to ignore. This has the benefit that you keep checking the other linters.
```suggestion
#' See [metadata Vlaanderen](https://metadata.vlaanderen.be/metadatacenter/srv/dut/catalog.search#/search?keyword=OGC:WCS) for more information # nolint line_length_liner;
```

File: R/get_feature_wfs.R
Comment (ThierryO) [inline R/get_feature_wfs.R:146]: You can omit the else when using early returns. That makes the code IMHO easier to read. Flipping the test makes the code shorter too and brings short exception forward.

Example
```
if (testA == "A") {
  if (testB == "a") {
    return("Aa")
  } else {
    return("Ab")
  }
} else {
  return("B")
}
```
Alternative version
```
if (testA != "A") {
  return("B")
}
if (testB == "a") {
  return("Aa")
}
return("Ab")
```

File: R/get_feature_wfs.R
Comment (ThierryO) [inline R/get_feature_wfs.R:149]: maybe add a `store_as_gml()` function with S3 methods for `xml_document` and `raw`.

File: R/get_feature_wfs.R
Comment (ThierryO) [inline R/get_feature_wfs.R:151]: Import `write_xml()`
```suggestion
        write_xml(content, destfile)
```

File: R/get_feature_wfs.R
Comment (ThierryO) [inline R/get_feature_wfs.R:161]: Start newline after pipe.

File: DESCRIPTION
Comment (florisvdh) [reply DESCRIPTION:21]: I'm using it with curly braces, which the base pipe doesn't support:

https://github.com/inbo/inbospatial/blob/e2bc5155cec668fe0b470458cf48baaec8227c11/R/add_wms.R#L432-L445

But I might change it indeed and use the lambda notation to keep things short.

```r
setting <- TRUE
mtcars |> 
  (\(x) if (setting) x else subset(x, cyl == 4))() |> 
  nrow()
```

Although that still reads more akward IMHO because of the extra parentheses; compared to:

```r
library(magrittr)
mtcars %>%
  {if (setting) . else subset(., cyl == 4)} |> 
  nrow()
```

File: R/crs.R
Comment (ThierryO) [reply R/crs.R:35]: It replaces `Floris Vanderhaeghe` with `[Floris Vanderhaeghe](https://github.com/florisvdh)` on the pkgdown website.

File: R/get_coverage_wcs.R
Comment (florisvdh) [reply R/get_coverage_wcs.R:23]: You must add a colon right after `nolint`, otherwise all linters will still be excluded. Also the linters must be comma-separated and at the end there must be a point AFAIK.

Comment (florisvdh) [reply R/get_coverage_wcs.R:23]: `# nolint: line_length_liner, xxxx.`

Comment (ThierryO) [reply R/get_coverage_wcs.R:23]: I've updated my suggestion.

File: R/get_feature_wfs.R
Comment (hansvancalster) [reply R/get_feature_wfs.R]: done in [e8b82f6](https://github.com/inbo/inbospatial/pull/1/commits/e8b82f643d3eb02451ebe57b6f14e3c36217155b)

File: R/get_feature_wfs.R
Comment (hansvancalster) [reply R/get_feature_wfs.R]: see [efa54d6](https://github.com/inbo/inbospatial/pull/1/commits/efa54d606d348cbb4514f97dbd014021634ce692)

File: R/get_feature_wfs.R
Comment (hansvancalster) [reply R/get_feature_wfs.R]: see [efa54d6](https://github.com/inbo/inbospatial/pull/1/commits/efa54d606d348cbb4514f97dbd014021634ce692)

File: R/get_feature_wfs.R
Comment (hansvancalster) [reply R/get_feature_wfs.R]: see [efa54d6](https://github.com/inbo/inbospatial/pull/1/commits/efa54d606d348cbb4514f97dbd014021634ce692)

File: R/get_coverage_wcs.R
Comment (hansvancalster) [reply R/get_coverage_wcs.R]: see [efa54d6](https://github.com/inbo/inbospatial/pull/1/commits/efa54d606d348cbb4514f97dbd014021634ce692)

File: R/crs.R
Comment (hansvancalster) [reply R/crs.R:35]: see [2de1bc2](https://github.com/inbo/inbospatial/pull/1/commits/2de1bc2d03ee829dbc36e8e4f48ff136eee23598)

File: R/get_feature_wfs.R
Comment (ThierryO) [inline R/get_feature_wfs.R:163]: Maybe set the default of the `destfile` argument. Then you can omit this line.

```
store_as_gml.raw <- function(content, destfile = tempfile(fileext = "gml"), ...) {
```

Comment (hansvancalster) [reply R/get_feature_wfs.R]: done!

Comment (hansvancalster) [reply R/get_feature_wfs.R]: Oops, that was not possible b/c destfile object is used further down in the code

Comment (ThierryO) [reply R/get_feature_wfs.R]: > Oops, that was not possible b/c destfile object is used further down in the code

You can solve that by having `store_as_gml()` return the `destfile` object.

**Comment by hansvancalster** [comment]
@ThierryO or @florisvdh anything else that needs to be done? I would like to see your approval early today, so that this version of the package can be used in coding club tomorrow.

**Comment by florisvdh** [comment]
> @ThierryO or @florisvdh anything else that needs to be done? I would like to see your approval early today, so that this version of the package can be used in coding club tomorrow.

OK, looking at it.

**Review by florisvdh — APPROVED** [review]
Great work @hansvancalster :tada:.

I added some suggestions in PR #9.

I much like the vignette, cool! It's nice that you show the **sf** way as well; very didactical!

Remains to be looked at:

- you're using `@family` with a string identifier for reference in `_pkgdown.yaml`. For such purpose, better use `@concept` instead. It's implemented in PR #9. It will not create a 'See also' section in the function docs. Otherwise you get _See also: 'Other web_services: ....', See also: 'Other crs: ...'._
  - my personal pref is to also have 'see also'. You can find a suggestion in the `family` branch that caters for both; feel free to use.
- in running `pkgdown::build_site()`, there's a warning, it could be suppressed with the option (GHA) or handled otherwise in the code perhaps:

```
Warning message:
The vignette title specified in \VignetteIndexEntry{} is different from the title in the YAML metadata. The former is "wfs_wcs", and the latter is "Example use cases to get data from web feature or web coverage services.". If that is intentional, you may set options(rmarkdown.html_vignette.check_title = FALSE) to suppress this check. 
```

**Comment by florisvdh** [comment]
I've rebased the `family` branch on `minor_updates` to prevent conflicts if you like to merge it after merging PR #9.

**Comment by hansvancalster** [comment]
> * my personal pref is to also have 'see also'. You can find a suggestion in the `family` branch that caters for both; feel free to use.

I don't see a PR that does that?

**Comment by florisvdh** [comment]
> I don't see a PR that does that?

No, but you can add it to minor_updates and push to PR #9 if you like (or to PR #1 after merging the latter):

```bash
git fetch --prune
git checkout --track origin/minor_updates
git merge --no-ff origin/family
git push
```

**Comment by florisvdh** [comment]
You can also have a look at https://github.com/inbo/inbospatial/compare/minor_updates..family

**Pull request merged**
